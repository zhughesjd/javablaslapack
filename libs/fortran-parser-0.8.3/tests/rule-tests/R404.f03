! kind_selector can have an expr.  there are too many to test all of them, 
! but we can test some common ones.
! Testing kind selector.
integer (kind = 4), parameter :: i
integer (kind = 4) :: p
integer (kind = foo) :: bar
integer(8) :: j
integer(kind(4)) :: k
integer(kind=i) :: m
integer(kind(j)) :: n
integer(i) :: o
integer(selected_int_kind(12)) :: d
real * 7 :: k
end
