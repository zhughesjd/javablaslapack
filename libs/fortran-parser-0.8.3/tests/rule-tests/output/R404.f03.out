********************************************
args[1]: R404.f03
R1101:main-program__begin:
R 501:type-declaration-stmt__begin:
R 406:int-literal-constant: digitString=[@13,153:153='4',<10>,4:16] kindParam=null
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 503:attr-spec: attr=813
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@21,170:170='i',<206>,4:33]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=1
R 501:type-declaration-stmt__begin:
R 406:int-literal-constant: digitString=[@30,188:188='4',<10>,5:16] kindParam=null
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@35,194:194='p',<206>,5:22]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 613:part-ref: id=[@44,212:214='foo',<206>,6:16] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@49,220:222='bar',<206>,6:24]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 406:int-literal-constant: digitString=[@53,232:232='8',<10>,7:8] kindParam=null
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@58,238:238='j',<206>,7:14]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 619:section-subscript-list__begin:
R 406:int-literal-constant: digitString=[@64,253:253='4',<10>,8:13] kindParam=null
R 619:section-subscript: hasLowerBound=true hasUpperBound=false hasStride=false isAmbiguous=true
R 619:section-subscript-list: count=1
R 613:part-ref: id=[@62,248:251='kind',<206>,8:8] hasSelectionSubscriptList=true hasImageSelector=false
R 612:data-ref: numPartRef=1
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@70,260:260='k',<206>,8:20]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 613:part-ref: id=[@76,275:275='i',<206>,9:13] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@81,281:281='m',<206>,9:19]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 619:section-subscript-list__begin:
R 613:part-ref: id=[@87,296:296='j',<206>,10:13] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 619:section-subscript: hasLowerBound=true hasUpperBound=false hasStride=false isAmbiguous=true
R 619:section-subscript-list: count=1
R 613:part-ref: id=[@85,291:294='kind',<206>,10:8] hasSelectionSubscriptList=true hasImageSelector=false
R 612:data-ref: numPartRef=1
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@93,303:303='n',<206>,10:20]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 613:part-ref: id=[@97,313:313='i',<206>,11:8] hasSelectionSubscriptList=false hasImageSelector=false
R 612:data-ref: numPartRef=1
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@102,319:319='o',<206>,11:14]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 619:section-subscript-list__begin:
R 406:int-literal-constant: digitString=[@108,347:348='12',<10>,12:26] kindParam=null
R 619:section-subscript: hasLowerBound=true hasUpperBound=false hasStride=false isAmbiguous=true
R 619:section-subscript-list: count=1
R 613:part-ref: id=[@106,329:345='selected_int_kind',<206>,12:8] hasSelectionSubscriptList=true hasImageSelector=false
R 612:data-ref: numPartRef=1
R 404:kind-selector: hasExpression=true typeSize=null
R 403:intrinsic-type-spec: type=400 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@114,355:355='d',<206>,12:34]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 501:type-declaration-stmt__begin:
R 404:kind-selector: hasExpression=false typeSize=[@120,364:364='7',<10>,13:7]
R 403:intrinsic-type-spec: type=401 hasKindSelector=true
R 504:entity-decl-list__begin:
R 504:entity-decl: id=[@124,369:369='k',<206>,13:12]
R 504:entity-decl-list: count=1
R 501:type-declaration-stmt: numAttributes=0
R 204:specification-part: numUseStmts=0 numImportStmts=0 numDeclConstructs=10
R9999:end-of-stmt:
R1103:end-program-stmt: id=null
R1101:main-program: hasProgramStmt=false hasExecutionPart=false hasInternalSubprogramPart=false
Parser exiting normally
********************************************
